src/node_task_runner.cc

```diff
@@ -291,57 +291,180 @@ void RunTask(const std::shared_ptr<InitializationResultImpl>& result,
-    fprintf(
-        stderr, "Can't find \"scripts\" field in %s\n", path.string().c_str());
-    result->exit_code_ = ExitCode::kGenericUserError;
-    return;
-  }
-
-  // If the command_id is not found in the scripts object, throw an error.
-  std::string_view command;
-  if (auto command_error =
-          scripts_object[command_id].get_string().get(command)) {
-    if (command_error == simdjson::error_code::INCORRECT_TYPE) {
+  bool have_scripts = main_object["scripts"].get_object().get(scripts_object) ==
+                      simdjson::error_code::SUCCESS;
+
+  std::string exec_cmd;
+  if (have_scripts) {
+    std::string_view cmd_string;
+    auto err = scripts_object[command_id].get_string().get(cmd_string);
+    if (err == simdjson::error_code::SUCCESS) {
+      exec_cmd.assign(cmd_string);
+      ProcessRunner runner(result,
+                           path,
+                           command_id,
+                           exec_cmd,
+                           path_env_var,
+                           positional_args);
+      runner.Run();
+      return;
+    }
+    if (err == simdjson::error_code::INCORRECT_TYPE) {
       fprintf(stderr,
               "Script \"%.*s\" is unexpectedly not a string for %s\n\n",
               static_cast<int>(command_id.size()),
               command_id.data(),
               path.string().c_str());
-    } else {
-      fprintf(stderr,
-              "Missing script: \"%.*s\" for %s\n\n",
-              static_cast<int>(command_id.size()),
-              command_id.data(),
-              path.string().c_str());
-      fprintf(stderr, "Available scripts are:\n");
-
-      // Reset the object to iterate over it again
-      scripts_object.reset();
-      simdjson::ondemand::value value;
-      for (auto field : scripts_object) {
-        std::string_view key_str;
-        std::string_view value_str;
-        if (!field.unescaped_key().get(key_str) && !field.value().get(value) &&
-            !value.get_string().get(value_str)) {
+      result->exit_code_ = ExitCode::kGenericUserError;
+      return;
+    }
+  }
+
+  // Try "bin"
+  simdjson::ondemand::value bin_value;
+  bool have_bin =
+      main_object["bin"].get(bin_value) == simdjson::error_code::SUCCESS;
+
+  if (!have_scripts && !have_bin) {
+    fprintf(stderr,
+            "Can't find \"scripts\" or \"bin\" fields in %s\n",
+            path.string().c_str());
+    result->exit_code_ = ExitCode::kGenericUserError;
+    return;
+  }
+
+  if (have_bin) {
+    simdjson::ondemand::json_type bin_type;
+    if (!bin_value.type().get(bin_type)) {
+      std::string exec_rel;
+
+      if (bin_type == simdjson::ondemand::json_type::string) {
+        // "bin": "./cli.js"
+        std::string_view rel;
+        if (!bin_value.get_string().get(rel)) {
+          std::string_view pkg_name;
+          if (!main_object["name"].get_string().get(pkg_name) &&
+              pkg_name == command_id) {
+            exec_rel.assign(rel);
```

Comment:
I think you're confusing std::string and std::string_view. These lines are not needed.